operator/src/trustee.rs

```diff
@@ -127,61 +131,77 @@ pub async fn generate_kbs_configurations(
 
 pub async fn generate_reference_values(
     client: Client,
-    namespace: &str,
+    job_namespace: &str,
+    trustee_namespace: &str,
     name: &str,
+    pcrs_compute_image: &str,
+    pcrs_boot_image: &str,
 ) -> anyhow::Result<()> {
-    let reference_values_in_json = include_str!("reference-values-in.json");
-    let mut reference_values_in = match serde_json::from_str(reference_values_in_json)? {
-        serde_json::Value::Object(vals) => vals,
-        _ => return Err(anyhow!("Reference values had unexpected shape")),
+    let job_name = "compute-pcrs";
+    let volume_name = "image";
+    let mountpoint = PathBuf::from("/image");
+
+    let mut flags = Vec::new();
+    let mut add_flag = |flag: &str, value: &str| {
+        flags.push(format!("--{flag}"));
+        flags.push(value.to_string());
     };
-    reference_values_in.insert(
-        "svn".to_string(),
-        serde_json::Value::String("1".to_string()),
-    );
-    let reference_values = reference_values_in
-        .iter()
-        .map(|(name, value)| {
-            if let serde_json::Value::String(hex) = value
-                && hex
-                    .chars()
-                    .all(|c| (c >= '0' && c <= '9') || (c >= 'a' && c <= 'f'))
-            {
-                Ok(ReferenceValue {
-                    version: "0.1.0".to_string(),
-                    name: format!("tpm_{name}"),
-                    expiration: chrono::DateTime::<chrono::Utc>::MAX_UTC,
-                    value: serde_json::Value::Array(vec![value.clone()]),
-                })
-            } else {
-                Err(anyhow!("Reference value '{value}' had unexpected shape"))
-            }
-        })
-        .collect::<Result<Vec<_>, _>>()?;
-    let reference_values_json = serde_json::to_string(&reference_values)?;
-
-    let mut data = BTreeMap::new();
-    data.insert(
-        "reference-values.json".to_string(),
-        reference_values_json.to_string(),
-    );
+    for (flag, path_suffix) in [
+        ("kernels", "usr/lib/modules"),
+        ("esp", "usr/lib/bootupd/updates"),
+    ] {
+        let full_path = mountpoint.join(path_suffix);
+        add_flag(flag, full_path.to_str().unwrap());
+    }
+    for (flag, value) in [("configmap", name), ("namespace", trustee_namespace)] {
+        add_flag(flag, value);
+    }
 
-    let config_map = ConfigMap {
-        metadata: kube::api::ObjectMeta {
-            name: Some(name.to_string()),
-            namespace: Some(namespace.to_string()),
+    let pod_spec = PodSpec {
+        service_account_name: Some("cocl-operator".to_string()),
```

Comment:
I would create a separate service account for the job only with a separate Role to only be able to create and modify the config maps in the trustee namespace